tests/connectors/document_index/test_document_index.py

```diff
@@ -61,6 +66,29 @@ def collection_path(aleph_alpha_namespace: str) -> CollectionPath:
     )
 
 
+@fixture
+def random_index(
+    document_index: DocumentIndexClient, aleph_alpha_namespace: str
+) -> Iterator[tuple[IndexPath, IndexConfiguration]]:
+    name = "".join(random.choices(string.ascii_letters + string.digits + "-.", k=20))
+    chunk_size, chunk_overlap = sorted(
+        random.sample([0, 32, 64, 128, 256, 512, 1024], 2), reverse=True
+    )
+    embedding_type = random.choice(get_args(EmbeddingType))
+
+    index = IndexPath(namespace=aleph_alpha_namespace, index=name)
+    index_configuration = IndexConfiguration(
+        chunk_size=chunk_size,
+        chunk_overlap=chunk_overlap,
+        embedding_type=embedding_type,
+    )
+    try:
+        document_index.create_index(index, index_configuration)
+        yield index, index_configuration
+    finally:
+        document_index.delete_index(index)
```

Comment:
Be aware that this might not be enough to clean the indices in all cases. When the runs fully crash / are interrupted / killed, this may skip the cleanup step, leaving us with dangling repositories

Reply:
Added a cleanup function `_teardown` that runs after all test and eventually removes all indexes that are older than 1 hour. Also it might make sense to move the test collection to another namespace (out of scope of this issue).

For this, it was necessary to:
- add `DocumentIndexClient.list_indexes(namespace)`
- add a timestamp suffix to the random name

What do you think about this?

Reply:
looks robust enough.